javalib/src/main/scala/java/io/BufferedWriter.scala

```diff
@@ -0,0 +1,50 @@
+package java.io
+
+class BufferedWriter(out: Writer, sz: Int) extends Writer {
+
+  if (sz <= 0) throw new IllegalArgumentException("Buffer size <= 0")
+
+  def this(out: Writer) = this(out, 4096)
+
+  private val buffer: Array[Char] = new Array[Char](sz)
+  private var pos: Int = 0
+  private var closed: Boolean = false
+
+  def close(): Unit = if (!closed) {
+    flush()
+    out.close()
+    closed = true
+  }
+
+  def flush(): Unit = {
+    ensureOpen()
+    out.write(buffer, 0, pos)
+    out.flush()
+    pos = 0
+  }
+
+  def newLine(): Unit =
+    write(System.lineSeparator(), 0, System.lineSeparator().length)
+
+  override def write(c: Int): Unit =
+    write(Array(c.toChar), 0, 1)
```

Comment:
IIUC there is always at least one spot free in the buffer (otherwise it would be flushed).

So couldn't this be:

```scala
buffer(pos) = c.toChar
pos += 1
if (pos == len)
  flush()
```